File: src/baseDatos/Serializador.java
package baseDatos;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;
import java.io.PrintWriter;
import java.util.ArrayList;
import java.util.List;


import gestorAplicacion.*;
import gestorAplicacion.gente.Cliente;
import gestorAplicacion.gente.Empleado;
import gestorAplicacion.gente.Gerente;
import gestorAplicacion.restaurante.Caja;
import gestorAplicacion.restaurante.Orden;
import gestorAplicacion.restaurante.Platillo;
import gestorAplicacion.restaurante.horarios;
import gestorAplicacion.restaurante.ingredientes;

public class Serializador {
	/**
	 * Serializamos una lista por el nombre de la clase
	 * 
	 * @param <E>       el generico se usa para poder agredar las clases que se
	 *                  crearon
	 * @param lista     Una lista de objetos
	 * @param className El nombre de la clase que queremos usar como nombre del
	 *                  archivo
	 */

	public static <E> void serializar(ArrayList<E> list, String className) {// para Arraylist
		FileOutputStream fileOut;
		try {
			String path = System.getProperty("user.dir") + "/src/baseDatos/temp/" + className + ".txt";
			// se crea un fileoutputstream para saber donde serializar los archivos
			fileOut = new FileOutputStream(path);
			// Se crea un objeto output stream para poder escribir en el archivo
			ObjectOutputStream out = new ObjectOutputStream(fileOut);
			// Guardamos la lista de objetos
			out.writeObject(list);
			out.close();
			fileOut.close();
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		} catch (IOException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
		System.out.println("Serilizando " + className);
	}

	public static void serializarTodo() {
		// Serializador.serializar(Gerente.arr,"Gerente");
		Serializador.serializar(Empleado.lista_empleados, "Empleado");
		Serializador.serializar(ingredientes.lista_ingredientes, "ingredientes");

	}
}

File: Python/baseDatos/Serializador.py
import pickle

from Python.gestorAplicacion.gente.cliente import Cliente
from Python.gestorAplicacion.gente.empleado import Empleado
from Python.gestorAplicacion.gente.gerente import Gerente
from Python.gestorAplicacion.restaurante.ingredientes import Ingredientes

def serializar(cls,tipo):
    if tipo=="Cliente":
        pickfile = open("Python/baseDatos/temp/Cliente.txt","wb")
        serializa = Cliente._lista_socio
        pickle.dumps(serializa,pickfile)
        pickfile.close

    elif tipo=="Empleado":
         pickfile = open("Python/baseDatos/temp/Empleado.txt", "wb")
         serializa = Empleado._lista_empleado
         pickle.dumps(serializa,pickfile)
         pickfile.close
    
    elif tipo=="Gerente":
        pickfile = open("Python/baseDatos/temp/Gerente.txt", "wb")
        serializa = Gerente._list_gerente
        pickle.dumps(serializa, pickfile)
        pickfile.close
    
    elif tipo=="ingredientes":
         pickfile = open("Python/baseDatos/temp/ingredientes.txt", "wb")
         serializa = Ingredientes.getListaIngredientes
         pickle.dumps(serializa, pickfile)
         pickfile.close
    
    elif tipo=="NumEmple":
         pickfile = open("Python/baseDatos/temp/NumEmple.txt", "wb")
         serializa = Empleado.getNumero_empleados
         pickle.dumps(serializa, pickfile)
         pickfile.close